thrust/system/hip/detail/reduce.h

```diff
@@ -51,126 +51,113 @@ BEGIN_NS_THRUST
+                             BinaryFunction                                              binary_op);
 
 namespace hip_rocprim
 {
-namespace __reduce
-{
-    template <class Policy, class InputIt, class Size, class T, class BinaryOp>
-    T THRUST_HIP_RUNTIME_FUNCTION
-    reduce(Policy& policy, InputIt first, Size num_items, T init, BinaryOp binary_op)
+    namespace __reduce
     {
-        if(num_items == 0)
-            return init;
-
-        void*       d_temp_storage     = NULL;
-        size_t      temp_storage_bytes = 0;
-        hipStream_t stream             = hip_rocprim::stream(policy);
-        T*          d_ret_ptr          = NULL;
-        bool        debug_sync         = THRUST_HIP_DEBUG_SYNC_FLAG;
-
-        // Determine temporary device storage requirements.
-        hip_rocprim::throw_on_error(rocprim::reduce(d_temp_storage,
-                                                    temp_storage_bytes,
-                                                    first,
-                                                    d_ret_ptr,
-                                                    init,
-                                                    num_items,
-                                                    binary_op,
-                                                    stream,
-                                                    debug_sync),
-                                    "reduce failed on 1st step");
-
-        // Allocate temporary storage.
-        d_temp_storage = hip_rocprim::get_memory_buffer(policy, sizeof(T) + temp_storage_bytes);
-        hip_rocprim::throw_on_error(hipGetLastError(), "reduce failed to get memory buffer");
-
-        d_ret_ptr = reinterpret_cast<T*>(reinterpret_cast<char*>(d_temp_storage)
-                                         + temp_storage_bytes);
-
-        hip_rocprim::throw_on_error(rocprim::reduce(d_temp_storage,
-                                                    temp_storage_bytes,
-                                                    first,
-                                                    d_ret_ptr,
-                                                    init,
-                                                    num_items,
-                                                    binary_op,
-                                                    stream,
-                                                    debug_sync),
-                                    "reduce failed on 2nd step");
-
-        T return_value = hip_rocprim::get_value(policy, d_ret_ptr);
-
-        hip_rocprim::return_memory_buffer(policy, d_temp_storage);
-        hip_rocprim::throw_on_error(hipGetLastError(), "reduce failed to return memory buffer");
-
-        return return_value;
+        template <class Policy, class InputIt, class Size, class T, class BinaryOp>
+        T   THRUST_HIP_RUNTIME_FUNCTION
+            reduce(Policy& policy, InputIt first, Size num_items, T init, BinaryOp binary_op)
```

Comment:
The function alignment is not correct because of the THRUST_HIP_RUNTIME_FUNCTION attribute.

Reply:
> We also added clang format file and we would like to avoid to change the original thrust files to avoid the future conflict.
> 
> I will upstream our clang file (based on AMD format file) and I would like to have a discussion about this first.

Sounds good to me. Yeah, I will close this PR for now. Other projects were having formatting issues, so I thought I could try and change the formatting with rocThrust. I'll close this PR.